### ISSUE TITLE
Help: Glutun suddenly stopped working after working fine for months

### ISSUE THREAD

**Issue by inquam** [qdm12/gluetun#230]
**TLDR**: *Glutun has been working fine for months and all of a sudden it just stopped.*

The glutun docker container is used by other containers and has worked fine for a few months. One day I had no connection and upon investigating the glutun docker container appears to not be bale to do its job. If I connect to it and try to check connectivity by "ping 8.8.8.8" for instance I get "operation not permitted".

1. Is this urgent?

    - [x] Yes
    - [] No

2. What VPN service provider are you using?

    - [x] PIA
    - [ ] Mullvad
    - [ ] Windscribe
    - [ ] Surfshark
    - [ ] Cyberghost

3. What's the version of the program?

    **See the line at the top of your logs**

    `Running version latest built on 2020-08-31T02:03:31Z (commit 9dcc009)`

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

Log lines got reversed when copied from portainer, so read from bottom to top :)

```log
2020-08-31T10:24:26.819+0200	INFO	openvpn: TLS Error: TLS handshake failed,
2020-08-31T10:24:26.819+0200	INFO	openvpn: TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity),
2020-08-31T10:22:16.497+0200	INFO	openvpn: UDP link remote: [AF_INET]46.246.3.150:1197,
2020-08-31T10:23:16.586+0200	INFO	openvpn: TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity),
2020-08-31T10:23:16.586+0200	INFO	openvpn: TLS Error: TLS handshake failed,
2020-08-31T10:23:16.586+0200	INFO	openvpn: SIGUSR1[soft,tls-error] received, process restarting,
2020-08-31T10:23:26.587+0200	INFO	openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]46.246.3.150:1197,
2020-08-31T10:23:26.587+0200	INFO	openvpn: UDP link local: (not bound),
2020-08-31T10:23:26.587+0200	INFO	openvpn: UDP link remote: [AF_INET]46.246.3.150:1197,
2020-08-31T10:22:16.497+0200	INFO	openvpn: UDP link local: (not bound),
2020-08-31T10:22:16.497+0200	INFO	openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]46.246.3.150:1197,
2020-08-31T10:22:16.497+0200	INFO	openvpn: CRL: loaded 1 CRLs from file [[INLINE]],
2020-08-31T10:22:16.492+0200	INFO	openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10,
2020-08-31T10:22:16.492+0200	INFO	openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020,
2020-08-31T10:22:16.489+0200	INFO	openvpn configurator: starting openvpn,
2020-08-31T10:22:16.486+0200	INFO	firewall: setting VPN connections through firewall...,
2020-08-31T10:22:16.486+0200	INFO	dns configurator: using DNS address 1.1.1.1 system wide,
2020-08-31T10:22:16.486+0200	INFO	dns configurator: using DNS address 1.1.1.1 internally,
2020-08-31T10:22:16.486+0200	INFO	Launching standard output merger,
2020-08-31T10:22:16.485+0200	INFO	dns over tls: falling back on plaintext DNS at address 1.1.1.1,
2020-08-31T10:22:16.485+0200	INFO	http server: listening on 0.0.0.0:8000,
2020-08-31T10:22:16.484+0200	INFO	routing: adding 192.168.1.0/24 as route via 172.18.0.1 eth0,
2020-08-31T10:22:16.480+0200	INFO	firewall: setting allowed subnets through firewall...,
2020-08-31T10:22:16.480+0200	INFO	firewall: enabled successfully,
2020-08-31T10:22:16.459+0200	INFO	firewall: enabling...,
2020-08-31T10:22:16.459+0200	INFO	openvpn configurator: checking for device /dev/net/tun,
2020-08-31T10:22:16.459+0200	INFO	routing: local subnet found: 172.18.0.0/16,
2020-08-31T10:22:16.459+0200	INFO	routing: default route found: interface eth0, gateway 172.18.0.1,
2020-08-31T10:22:16.429+0200	INFO	storage: Merging by most recent 6011 hardcoded servers and 6011 servers read from /gluetun/servers.json,
Version information: enabled,
Public IP check period: 12h0m0s,
TinyProxy settings: disabled,
ShadowSocks settings: disabled,
 |--VPN input ports: ,
 |--Allowed subnets: 192.168.1.0/24,
Firewall settings:,
 |--Keep nameserver (disabled blocking): no,
 |--Update: every 24h0m0s,
 |--IPv6 resolution: disabled,
 |--Verbosity details level: 0/4,
 |--Validation log level: 0/2,
 |--Verbosity level: 1/5,
  |--::ffff:0:0/96,
  |--fe80::/10,
  |--fc00::/7,
  |--::1/128,
  |--169.254.0.0/16,
  |--192.168.0.0/16,
  |--172.16.0.0/12,
  |--10.0.0.0/8,
  |--127.0.0.1/8,
 |--Private addresses:,
 |--Allowed hostnames:,
  |--,
 |--Block ads: disabled,
 |--Block surveillance: disabled,
 |--Caching: enabled,
 |--Block malicious: enabled,
  |--cloudflare,
 |--DNS over TLS provider:,
DNS over TLS settings:,
|--IP Status filepath: /tmp/gluetun/ip,
|--Group ID: 1000,
|--Timezone: europe/stockholm,
|--User ID: 1000,
System settings:,
 |--Encryption preset: strong,
 |--Network protocol: udp,
 |--Region: sweden,
|--Private Internet Access settings:,
|--Run as root: no,
|--Verbosity level: 1,
|--Password: [redacted],
|--User: [redacted],
OpenVPN settings:,
2020-08-31T10:22:16.400+0200	INFO	Settings summary below:,
2020-08-31T10:22:16.400+0200	INFO	TinyProxy version: 1.10.0,
2020-08-31T10:22:16.381+0200	INFO	IPtables version: v1.8.4,
2020-08-31T10:22:16.378+0200	INFO	OpenVPN version: 2.4.9,
2020-08-31T10:22:16.380+0200	INFO	Unbound version: 1.10.1,
💸  Help me? https://github.com/sponsors/qdm12,
☕  Slack? Join from the Slack button on Github,
💻  Email? quentin.mcgaw@gmail.com,
🔧  Need help? https://github.com/qdm12/gluetun/issues/new,
📣  Persistent server IP addresses at /gluetun/servers.json, please BIND MOUNT,
Running version latest built on 2020-08-31T02:03:31Z (commit 9dcc009),
```

Configuration file:

The only thing added to the config after it stopped working to test is the line "privileged: true". So initial working config did not have this line and now it doesn't work with or without.

```yml
# Gluetun VPN
gluetun:
  image: qmcgaw/private-internet-access
  container_name: gluetun
  privileged: true
  cap_add:
    - NET_ADMIN
  networks:
    - t2_proxy
  ports:
    - 8888:8888/tcp # Tinyproxy
    - 8388:8388/tcp # Shadowsocks
    - 8388:8388/udp # Shadowsocks
    - 8000:8000/tcp # Built-in HTTP control server
  volumes:
    - /docker/shared:/shared
  # command:
  environment:
    # More variables are available, see the readme table
    VPNSP: private internet access
    PROTOCOL: "udp"
    OPENVPN_VERBOSITY: 1
    OPENVPN_ROOT: "no"
    OPENVPN_TARGET_IP:
    TZ: $TZ
    PUID: $PUID
    PGID: $PGID

    PORT_FORWARDING_STATUS_FILE: /shared/pia_forwarded_port

    # PIA, Windscribe and Surfshark only
    REGION: Sweden
    USER: $VPN_USERNAME
    PASSWORD: $VPN_PASSWORD

    # PIA only
    PIA_ENCRYPTION: "strong"
    PORT_FORWARDING: "on"

    # Mullvad only
    COUNTRY: "Sweden"
    CITY:
    ISP:

    # Mullvad and Windscribe only
    PORT:

    # DNS over TLS
    DOT: "on"
    DOT_PROVIDERS: "cloudflare"
    DOT_IPV6: "off"
    DOT_VERBOSITY: 1
    BLOCK_MALICIOUS: "on"
    BLOCK_SURVEILLANCE: "off"
    BLOCK_ADS: "off"
    UNBLOCK:
    DNS_UPDATE_PERIOD: "24h"
    # Firewall
    EXTRA_SUBNETS: "192.168.1.0/24"
    # Shadowsocks
    SHADOWSOCKS: "off"
    SHADOWSOCKS_PASSWORD:
    # Tinyproxy
    TINYPROXY: "off"
    TINYPROXY_USER:
    TINYPROXY_PASSWORD:
  restart: always
```

I then have other containers that use

```yml
depends_on:
  - gluetun
network_mode: service:gluetun
```

Host OS: Ubuntu 20.04

**Comment by qdm12** [comment]
The error is most likely a bad IP address (although I just updated them). Can you try with another region?

Also, try with `VPNSP: "private internet access old"`. By default it now uses the newer PIA servers which may be not functional.

For the ping, that's expected I think as you need to be super root privileged on the host and all that as I recall to ping 😄

Finally, beware PUID and PGID don't do anything, these are specific to linuxserver's Docker images I think. If you really want to set them for subprocesses (openvpn, tinyproxy, files written) you need to use UID and GID. They both default to `1000`.

**Comment by inquam** [comment]
Tried with the "old" and another region and it worked. Thank you. A small side note, I always write the forwarded port to a file, then have to look in that file and update the config of one container every time gluetun goes down. Is there a way to configure gluetun so it always presents one port and mapps that to the correct dynamic one internally?

**Comment by qdm12** [comment]
Alright cool, I'll double check the IP address for Sweden and the new servers. Maybe it's wrong on their end too 😕 Let's keep the issue opened.

For the forwarded port, I don't think there is a way to change the fact that the port is a dynamic number on PIA's end (the VPN server). There are some scripts some users shared in other issues in this repo, to dynamically update the forwarded port in other containers like rutorrent. You could also get the forwarded port using the http control server (see the wiki about the control server).

**Comment by inquam** [comment]
Ahh ok, thought gluetun might be able to do a port forward from a defined port to the dynamic port assigned by PIA. I had something going where I changed the "command" parameter of the container needing the port to have it check for the file, update the config and then start the program (qbittorrent) but didn't feel very elegant :)

**Comment by qdm12** [comment]
There is #205 but I doubt it will solve the issue. Also the advantage of using the http control server in a script is that you can just call http://localhost:8080/openvpn/forwarded_port (from the top of my head) as it's sharing the same network host 😉 A bit simpler than the file approach.

**Comment by inquam** [comment]
Ok, so do I interpret it correct that this commit would make the PIA forwarded port available on port 9000? But the commit is not part of a release yet? So for instance Qbittorrent could always use port 9000 no matter which actual port PIA provides? If so. perhaps add an option to the docker-compose file to be able to choose port?

**Comment by qdm12** [comment]
@inquam Please see #203 I have a question for you!

**Comment by qdm12** [comment]
The Sweden IP address PIA specified in their file doesn't work for some reason. I'll keep it that way to avoid crazy updating logic, they may fix their server sometime.
